### PR TITLE
st61458

### PR DESCRIPTION
**Pull request by NaraaB**
Please accept the pull request for project NNPTP20-1

### CODE REVIEW

File: INPTPZ1/ComplexNumber.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        public double RealPart { get; set; }
+        public double ImaginaryPart { get; set; }
+
+        public readonly static ComplexNumber Zero = new ComplexNumber()
+        {
+            RealPart = 0,
+            ImaginaryPart = 0
+        };
+        public bool isComplexZero()
```
Comment (rodi0878) [inline INPTPZ1/ComplexNumber.cs]: Doesn't comform to C# naming guidelines

File: INPTPZ1/ComplexNumber.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        public override bool Equals(object obj)
+        {
+            if (obj is ComplexNumber)
+            {
+                ComplexNumber a = obj as ComplexNumber;
+                return a.RealPart == RealPart && a.ImaginaryPart == ImaginaryPart;
+            }
+            return base.Equals(obj);
+        }
+        public ComplexNumber Multiply(ComplexNumber b)
```
Comment (rodi0878) [inline INPTPZ1/ComplexNumber.cs]: From method signature "b" is not a clear name

File: INPTPZ1/Program.cs [hunk 3]
```diff
@@ -23,274 +21,153 @@ namespace INPTPZ1
...
+                ComplexNumber.Zero,
+                ComplexNumber.Zero,
+                new ComplexNumber { RealPart = 1},
+            });
+            derivationOfPolynome = polynome.Derivate();
+            Console.WriteLine("f(x) = " + polynome);
+            Console.WriteLine("f'(x) = " + derivationOfPolynome);
+        }
+
         static void Main(string[] args)
```
Comment (rodi0878) [inline INPTPZ1/Program.cs:52]: Too long and too much complex method

File: INPTPZ1/Program.cs [hunk 4]
```diff
@@ -23,274 +21,153 @@ namespace INPTPZ1
...
-                    Imaginari = a.Imaginari + b.Imaginari
-                };
-            }
-            public double GetAngleInDegrees()
-            {
-                return Math.Atan(Imaginari / Re);
+        private static void FindSolutionByNewton()
+        {
+            iteration = 0;
+            for (int q = 0; q < 30; q++)
```
Comment (rodi0878) [inline INPTPZ1/Program.cs]: why q? whats q?

**Review by rodi0878 — CHANGES_REQUESTED** [review]
There are few things that needs to be resolved before merging. 
Also there are errors in some file manipulations:
.gitignore - you must not delete it
.vs/INPTPZ1/v14/.suo - must not be added to repo

Please fix these errors in this PR and then I will rereview it

**Comment by NaraaB** [comment]
Please take a look at the new commit

**Comment by rodi0878** [comment]
Main() is still too long and too much complex.

**Comment by NaraaB** [comment]
Please take a look at the new commit, where the main method is reduced in size

File: INPTPZ1/Program.cs (hunk 3 above)
Comment (NaraaB) [reply INPTPZ1/Program.cs:52]: reduced size of main method by introducing new method

**Comment by rodi0878** [comment]
Accepted

**Pull request closed without merging**